### DISCUSSION TITLE
BOM List

### DISCUSSION THREAD

**Discussion by fysnet** [ataradov/usb-sniffer#4] [General]
Hi,

Unless I am missing something, I don't see a BOM. Do you have a list and if not, will you please post one?

Thank you,
Ben

**Comment by ataradov** [comment]
I'll see what kind of BOM I can extract from the KiCad. I don't know if it would be too useful, since most generic components (like resistors and capacitors) won't have exact part number. And there are 5 major components that have no substitutions, everything else can be substituted based on availability.

**Comment by ataradov** [comment]
BOM was added https://github.com/ataradov/usb-sniffer/blob/main/bin/usb-sniffer-bom.csv

**Comment by fysnet** [comment]
Thank you.  That makes it easier to find what is needed.

Might I suggest that you include a voltage/current rating for the LEDs since the associated resistors will rely on this rating.

For example, the USB gives 5v0 and if the LED was 2v6 at .03 amps, you would need an 80ohm current limiting resister. Maybe a note or something stating that the current BOM and/or other specifications expect an LED at so-many volts/current with the listed resisters.

Hopefully, whomever is building this for themselves will know this and modify your BOM accordingly for the LEDs they choose.

Thank you again. This looks to be an interesting project I might build myself.

Ben

**Comment by ataradov** [comment]
It literally does not matter, pick whatever you like. Those theoretical current calculations for modern LEDs are pretty pointless. They are plenty bright for indication purposes at  single digit uA currents. And the brightness is a personal preference.

Modern LED as 30 mA would burn your eyes out. I usually put 10-20 kOhm if the board to be used as a bare board without the case. On my board I lowered the value to 1 kOhm because I wanted them to be a bit brighter to be better visible though the 3D printed case. And with that value, I can't look directly at them for more than a couple seconds.

The values on the schematic should be workable, and if they don't provide enough brightness for some specific LED, the resistor value can be adjusted. But that adjustment is mostly a trial and error anyway.

**Comment by Sephrael** [comment]
Thank you so much for putting in the extra time to make the BOM. I don't have an electrical engineering background, but have a very strong hobbyist interest in all sorts of things and do have some familiarity with the general principles and fundamentals of electronics. (though am nowhere near the field professionally)
I was confused by how I was supposed to figure out where the rest of the parts were and was about to post and ask about it(honestly didn't even know what BOM stood for initially lol) vs possibly giving up. Anyways, I don't have the background nor any spare parts to speak of and would have been overwhelmed without having some kind of starting point to just try to guess what everything was from the picture. Just wanted to say how helpful it was that this is available for dummies like me!

**Comment by Sephrael** [comment]
I would just add it would be totally worth adding the csv to the main github repository and the readme(unless its there and I missed it)

**Comment by ataradov** [comment]
BOM is in the main repo. Second reply in this thread - https://github.com/ataradov/usb-sniffer/blob/main/bin/usb-sniffer-bom.csv

Also, you don't figure out components from the picture, you look at the schematic, which is also in the repo - https://github.com/ataradov/usb-sniffer/blob/main/bin/usb-sniffer.pdf

**Reply by Sephrael** [reply]
Ahh gotcha. I should have been more clear, I saw the csv in this thread for the first time because I had the same question as OP. I probably didn’t recognize the acronym for Bill of Materials when I was looking through the repo for a Parts List and there wasn’t a mention about it in the instructions/readme. Honestly though, that’s probably just a result of my inexperience and venturing further out of my expertise than is probably wise, but oh well I think I’m going to try building this. It combines a lot of things I enjoy, expanding my knowledge, using my 3d printer and helping me troubleshoot another thing I’m working on. I’m mostly worried about my soldering competence with the smaller pads and shorting the chips but there’s only one way to learn 🤷‍♂️. I’m also very curious to see how the GPT vision model handles my questions on the schematics how well it teaches the practical application of topics I have a limited understanding about. 

Anyhow, thanks again!

**Comment by GsmHack4you** [comment]
hi i try to order this project on PCBWay but they not accept BOM file response from it this email 

Hello ,
Thanks for your inquiring on PCBWay, this is Lynne.
ForPCBWay product No. T-N2W663927A.
For the common resistors and capacitors, we can recommend the part number, but not for the other components.
Pls add the manufacture part number into the bom-sheet to help confirm the correct/final components.
Looking forward to receive new BOM attach by email to move ahead the quotation, thanks.

Below link of BOM sample for your reference.
https://www.pcbway.com/img/images/pcbway/Sample_BOM_PCBWay.xlsx

do  someone can help me to make correct BOM file with name of parts from PCBWay database?

**Comment by ataradov** [comment]
This BOM is not designed for one-click orders. It is there to simplify ordering generic components, you still need to source other unique components. It is really targeted for hand assembly.

On a quick look, PcbWay library does not have two main components - FPGA and MCU. There is not much you can do about it.

**Reply by GsmHack4you** [reply]
Is it possible to order a ready-made device somewhere? In theory, I could assemble them myself, but in practice I would prefer to buy a ready-made product whose operation I can be sure of

**Reply by ataradov** [reply]
Not that I'm aware of. I don't manufacture them myself.
